oci/spec_opts.go

```diff
@@ -106,6 +106,43 @@ func setCPU(s *Spec) {
 	}
 }
 
+// WithCPUShares sets the container's cpu shares
+func WithCPUShares(shares uint64) SpecOpts {
+	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
+		setCPU(s)
+		s.Linux.Resources.CPU.Shares = &shares
```

Comment:
These are Linux specific features. Why move from `spec_opts_linux.go` to `spec_opts.go` ?

Reply:
Sorry, I should have added some context 🙂. Added some explanation [`here`](https://github.com/containerd/containerd/pull/4590#issue-494457733)

Reply:
Thanks for the input.
OCI spec defines different configuration depending on platform: https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L27 So there may be different set of `WithXYZ` APIs for Mac/Linux/Window. 
So Linux specific routines should remain in `spec_opts_linux.go`.
Also Mac/Linux are not entirely same from containerd perspective, currently containerd doesn't support Mac platform (even though its possible to build it).

Reply:
It's probably fine to do this, but with namespacing for the methods that are not behind a build flag.... e.g. `WithLinuxCPUShares`.

I think the idea here is to be able to do spec generation without having to be on the target platform?

Reply:
@mxpv Thanks for the quick response! Just to be clear, I am not asking about building and running `containerd` on `Mac`. For example, in our environment, both [`nomad-driver-containerd`](https://github.com/Roblox/nomad-driver-containerd) and `containerd-1.3.4` runs on an  `Ubuntu 18.04` environment, however, it would be nice that a developer could clone the repo `nomad-driver-containerd` and be able to do his/her development both on Mac/Linux. Hence my suggestion to keep these functions agnostic of the platform.

Reply:
Currently we have these opts separated per Linux/Window platform (even though windows specific ones are namespaced). I'm not sure that mixing them would be a good approach. 

It's still possible to generate Linux spec from Mac with a custom spec opt:
```go
func WithCustomConfig(shares uint64) SpecOpts {
	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
		/// Fill up the spec here.
		return nil
	}
}
```

Reply:
> be able to do his/her development both on Mac/Linux

You can specify linux tag and/or GOOS (I do development from Mac).

Reply:
@mxpv Thanks! `export GOOS=linux` works perfectly. I will close the PR.